R/mod_packageDependencies.R

```diff
@@ -0,0 +1,336 @@
+      
+      riskmetric_assess <-
+        riskmetric::pkg_ref(pkg_name,
+                            source = "pkg_cran_remote",
+                            repos = c("https://cran.rstudio.com")) %>%
+        dplyr::as_tibble() %>%
+        riskmetric::pkg_assess()
+      
+      riskmetric_score <-
+        riskmetric_assess %>%
+        riskmetric::pkg_score(weights = metric_weights)
+      
+      cli::cli_progress_update(id = id)
+      
+      return(list(name = riskmetric_assess$package, version = riskmetric_assess$version, score = round(riskmetric_score$pkg_score,2) ))
+    }
+    
+    # used for adding action buttons to data_table
+    shinyInput <- function(FUN, len, id, ...) {
+      inputs <- character(len)
+      for (i in seq_len(len)) {
+        inputs[i] <- as.character(FUN(paste0(id, i), ...))
+        # change icon from arrow to upload if not loaded into db yet
+        pkg_name <- pkg_df()[i, 3] %>% pull()
+        if (!pkg_name %in% loaded2_db()) {
+          inputs[i] <- gsub("fas fa-arrow-right fa-regular", "fas fa-upload fa-solid", inputs[i])
+        }
+      }
+      inputs
+    }
+    
+    depends <- reactive({
+      req(selected_pkg$name() != "-")
+      riskmetric::pkg_ref(selected_pkg$name()) %>% 
+        riskmetric::assess_dependencies() 
+    })
+    
+    revdeps <- reactive({
+      req(selected_pkg$name() != "-")
+      riskmetric::pkg_ref(selected_pkg$name()) %>% 
+        riskmetric::assess_reverse_dependencies() 
+    })
+    
+    dd <- eventReactive(pkg_df(), {
+      x <- deepdep::deepdep(selected_pkg$name(), depth = 2, dependency_type = c("Depends", "Imports", "LinkingTo"))
+      if (nrow(x) == 0) {
+        x <- data.frame(origin = selected_pkg$name(),
+                        name = pull(pkg_df()[1, 3]), version = NA_character_, type = "Imports",
+                        origin_level = 0L, dest_level = 1L)
+        attributes(x)$class <- c("deepdep", "data.frame")
+        attributes(x)$package_name <- selected_pkg$name()
+      }
+      return(x)
+    })
+    
+    tabready <- reactiveVal()
+    add_pkg <- reactiveVal()
+    
+    observeEvent(parent$input$tabs, {
+      tabready(0L)
+      if(parent$input$tabs == "Package Dependencies") {
+        tabready(1L)
+      }
+      
+    })
+    
+    m_id <- reactiveVal()
+    # start the progress message as soon as possible
+    observeEvent(list(selected_pkg$name(),tabready()), {
+      req(selected_pkg$name() != "-")
+      req(tabready() == 1L)
+      # this is really where the progress bar should start
+      m_id(cli::cli_progress_message("About to Create Data Table...", .auto_close = FALSE))
+    }, ignoreInit = TRUE)
+    
+    pkg_df <- eventReactive({selected_pkg$name(); tabready()}, {
+      req(!rlang::is_empty(selected_pkg$name()))
+      req(selected_pkg$name() != "-")
+      req(tabready() == 1L)
+
+      pkginfo <- riskmetric::pkg_ref(selected_pkg$name()) %>% 
+        riskmetric::assess_dependencies() %>%  
```

Comment:
We already have this info with reactive `depends()` on line 77.

Reply:
Right.  I replaced this piece of code with `depends()`